src/signup/signup.js

```diff
@@ -1,10 +1,75 @@
 import React from "react";
+import { Link } from "react-router-dom";
+import FormControl from '@material-ui/core/FormControl';
+import InputLabel from '@material-ui/core/InputLabel';
+import Input from '@material-ui/core/Input';
+import Paper from '@material-ui/core/Paper';
+import { withStyles } from '@material-ui/core/styles';
+import CssBaseline from '@material-ui/core/CssBaseline';
+import Typography from '@material-ui/core/Typography';
+import Button from '@material-ui/core/Button';
+import styles from "./styles"
+import { Form } from "reactstrap";
+const firebase = require("firebase");
 
 class SignupComponent extends React.Component {
     
     render() {
-        return(<div>Hello World from Signup</div>)
+
+        const { classes } = this.props;
```

Comment:
What's your intention here? This makes me assume that the parent component will pass all the styling classes down through props. Therefore the parent component will control all of the styling for this component... That might make this component less portable and self-contained. But I'm curious to see if there are advantages to doing it that I'm not aware of.

Reply:
Well, here classes property is not passed by the parent component. I'm using `withStyles` from Material-UI. You can see it at the end of the code: `export default withStyles(styles)(SignupComponent)`.  It injects classes to the props:
https://material-ui.com/guides/typescript/#usage-of-withstyles